ImageFeed/ViewControllers/ImagesList/ImageListViewController.swift

```diff
@@ -3,71 +3,84 @@ import SnapKit
 import Kingfisher
 import ProgressHUD
 
-final class ImageListViewController: UIViewController, ImageListViewControllerProtocol {    
+final class ImageListViewController: UIViewController, ImageListViewControllerProtocol {
+    
     override var preferredStatusBarStyle: UIStatusBarStyle { .lightContent }
-    private var profileInfoObserver: NSObjectProtocol?
     
-    var imagesListService: ImagesListServiceProtocol?
     var presenter: ImageListViewPresenterProtocol?
-    let imageListViews = ImageListViews()
+    var alertPresenter: AlertPresenter?
+    let imageListView = ImageListView()
+    let dateService = DateService()
+    var heightsOfRows: [String: CGFloat] = [:]
     
-    var heightsOfRows: [String:CGFloat] = [:]
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        setupTokenObserver()
-        presenter?.photosObserver()
-        
-        imageListViews.tableView.delegate = self
-        imageListViews.tableView.dataSource = self
+        imageListView.setTableViewDelegateAndDataSource(self)
+        presenter?.setupObservers()
         
         setTableView()
     }
     
-    private func setTableView() {
-        view.addSubview(imageListViews.tableView)
+    func requestFullSizeImage(viewController: SingleImageViewController, url: URL) {
+        UIBlockingProgressHUD.show()
+        viewController.imageView.kf.setImage(
+            with: url, placeholder: UIImage(named: "DownloadingImage")) { [weak self] result in
+                guard let self = self else { return }
+                DispatchQueue.main.async {
+                    switch result {
+                    case .success(let imageResult):
+                        UIBlockingProgressHUD.dismiss()
+                        viewController.rescaleAndCenterImageInScrollView(image: imageResult.image)
+                    case .failure(_:):
+                        UIBlockingProgressHUD.dismiss()
+                        self.alertPresenter?.presentErrorAlert()
+                    }
```

Comment:
Здесь можно лучше, чтобы не повторяться
```suggestion

                    UIBlockingProgressHUD.dismiss()
                    switch result {
                    case .success(let imageResult):
                        viewController.rescaleAndCenterImageInScrollView(image: imageResult.image)
                    case .failure:
                        self.alertPresenter?.presentErrorAlert()
                    }
```